frontend/src/__tests__/api/authors.test.js

```diff
@@ -0,0 +1,124 @@
+import '@testing-library/jest-dom';
+import { act, renderHook, waitFor } from '@testing-library/react';
+import {
+  useAddAuthor,
+  useGetAuthor,
+  useGetAuthors,
+  allAuthorsQuery,
+  anAuthorQuery,
+  addAuthorMutation
+} from '../../../api/authors';
+import {
+  aBookQuery,
+  addBookMutation,
+  allBooksQuery,
+  removeBookMutation,
+  useAddBook,
+  useGetBook,
+  useGetBooks,
+  useRemoveBook
+} from '../../../api/books';
+import { MockedProvider } from '@apollo/client/testing';
+
+describe('api Authors test', () => {
+  it('should get all authors', async () => {
+    const getAuthorsMock = {
+      request: {
+        query: allAuthorsQuery,
+        variables: {}
+      },
+      result: {
+        data: {
+          getAuthors: [
+            {
+              id: '1',
+              firstName: 'J.K.',
+              lastName: 'Rowling'
+            },
+            {
+              id: '2',
+              firstName: 'Dan',
+              lastName: 'Gookin'
+            }
+          ]
+        }
+      }
+    };
+    const wrapper = ({ children }) => (
+      <MockedProvider mocks={[getAuthorsMock]} addTypename={false}>
+        {children}
+      </MockedProvider>
+    );
+
+    const { result } = renderHook(() => useGetAuthors(), { wrapper });
+    await waitFor(() => expect(result.current.authors.length).not.toBe(0));
```

Comment:
I would just double check that this `waitFor` is actually required.

Reply:
I was not able to get it to work without waitFor. From https://testing-library.com/docs/dom-testing-library/api-async/  my understanding is it is required as the hook is called and has to resolve (which the waitFor allows). I do not know of any way around that.